core/scripts/cli/wallet.js

```diff
@@ -1,36 +1,98 @@
+// TODO:
+// Encrypt account data files
+// Show UMA token balance in wallet-info
+// Allow user to change directory in which wallets are stored
+// Allow more than one wallet backups
+// Connect to MetaMask
+
 const inquirer = require("inquirer");
+const os = require("os");
+
+// Wallet module helpers
+const createNewAccount = require("./wallet/createNewAccount");
+const createWalletDirectory = require("./wallet/createWalletDirectory");
+const loadSavedAccount = require("./wallet/loadSavedAccount");
+const restoreBackupAccount = require("./wallet/swapDefaultAndBackupAccounts");
+const readDefaultAccount = require("./wallet/readDefaultAccountInfo");
+const recoverAccount = require("./wallet/recoverAccount");
+const style = require("./textStyle");
+
+const ACTIONS = ["info", "init", "restore-backup", "recover", "help", "back"];
 
 const wallet = async () => {
   const prompts = [
     {
       type: "list",
       name: "walletTopMenu",
-      message: "Wallet actions",
-      choices: ["view wallet address", "wallet balance", "transfer tokens", "back"]
+      message: "UMA wallet actions",
+      choices: ACTIONS
     }
   ];
 
   return await inquirer.prompt(prompts);
 };
 
 module.exports = async function(web3) {
-  const accounts = await web3.eth.getAccounts();
+  // Get saved .uma directory for user (if it exists) and copy default account into web3.accounts.wallet
+  // @dev: Get home directory in platform agnostic way, src=https://stackoverflow.com/questions/9080085/node-js-find-home-directory-in-platform-agnostic-way
+  const homedir = os.homedir();
+  const umaDirectory = `${homedir}/.uma`;
+  const walletDirectory = createWalletDirectory(umaDirectory);
+  const accountDataFile = `${walletDirectory}/account.seed`;
+  loadSavedAccount(web3, accountDataFile);
+
   try {
-    const inputs = await wallet();
-    switch (inputs["walletTopMenu"]) {
-      case "view wallet address":
-        console.log(accounts[0]);
+    const inputs = (await wallet())["walletTopMenu"];
+    switch (inputs) {
+      // INFO: Display default account information for user
+      case ACTIONS[0]:
+        await readDefaultAccount(web3);
+        break;
+
+      // INIT: Create a new account for user
+      case ACTIONS[1]:
+        await createNewAccount(web3, accountDataFile);
+        break;
+
+      // RESTORE: Replace account backup with default account
+      case ACTIONS[2]:
+        await restoreBackupAccount(web3, accountDataFile);
+        break;
+
+      // RECOVER: Allow user to enter a private key for their desired default account
+      case ACTIONS[3]:
+        await recoverAccount(web3, accountDataFile);
```

Comment:
If we're going to store user private keys locally, we probably need to encourage them to back them up elsewhere, tell them where the file is, and tell them that it's not encrypted. I would also suggest using a mnemonic rather than a raw private key since users are much more accustomed to dealing with mnemonics. Wallet management is pretty scary, and I see a lot of risk vectors for users accidentally losing their private keys, so it might make sense for us to tell users to not put any real assets into wallets that are created here.

Reply:
See comment above